packages/components/hooks/use-scroll-position.ts

```diff
@@ -0,0 +1,29 @@
+import { useRef, useState } from 'react'
+
+import { useIsomorphicLayoutEffect } from './use-isomorphic-layout-effect'
+
+interface ScrollPosition {
+  ref: React.RefObject<HTMLDivElement>
+  scrollPos: number
+}
+
+const useScrollPosition = (): ScrollPosition => {
+  const ref = useRef<HTMLDivElement>(null)
+  const [scrollPos, setScrollPos] = useState(0)
+
+  useIsomorphicLayoutEffect(() => {
```

Comment:
I recommend against using non-null assertions for things that **_can_** happen. If a consumer of the hook forgets to set it correctly it is better to provide a more helpful message. 
```js
if (!ref.current) {
   throw new Error("useScrollPosition ref not set correctly")
   // OR
   console.warn("....")
   return
}
```

otherwise, it will just throw a cryptic error `Cannot read a property addEventListener of undefined`.